main.go

```diff
@@ -102,6 +104,10 @@ func main() {
 
 	ms := storage.NewDiskMetricStore(*persistenceFile, *persistenceInterval, prometheus.DefaultGatherer, logger)
 
+	if *pushEscaped {
+		model.NameValidationScheme = model.UTF8Validation
```

Comment:
Let's explicitly set `model.LegacyValidation` if the flag is not set (to defend against a change of the default in prometheus/client_model).